contracts/BodhiToken.sol

```diff
@@ -84,19 +92,23 @@ contract BodhiToken is StandardToken, Ownable {
   }
 
   /// @notice Allows contract owner to mint tokens at any time
-  /// @param _amount Amount of tokens to mint
+  /// @param _amount Amount of tokens to mint in lowest denomination of BOT
   function mintReservedTokens(uint256 _amount) onlyOwner {
     uint256 checkedSupply = totalSupply.add(_amount);
     require(checkedSupply <= tokenTotalSupply);
 
     mint(owner, _amount);
   }
 
-  /// @notice Shows the Ether to BOT exchange rate
-  /// @param _weiAmount Ether amount to convert
+  /// @notice Shows the amount of BOT the user will receive for amount of exchanged wei
+  /// @param _weiAmount Exchanged wei amount to convert
   /// @return The amount of BOT that will be received
   function getTokenExchangeAmount(uint256 _weiAmount) constant returns(uint256) {
-    return initialExchangeRate.mul(_weiAmount);
+    require(_weiAmount > 0);
```

Comment:
I know here `decimals` and `nativeDecimals` (`exchangeTokenDecimals`) are constant, and I don't think we should change it. However, I would expect to see some test cases to verify our formula is correct. One way to do it is to pass `decimals` and `nativeDecimals` into `getTokenExchangeAmount(_weiAmount, _decimals, _nativeDecimals)`, so that in our test script, we can then test on `getTokenExchangeAmount` such as:
`getTokenExchangeAmount(250, 18, 18) = 25000`
`getTokenExchangeAmount(2500000, 8, 12) = 25000`etc

Reply:
In addition, we need to also add `exchangeRate` as a parameter, so that we can have more test on exchangeRate as well.

Reply:
In addition, we need to also add exchangeRate as a parameter, so that we can have more test on exchangeRate as well.

Reply:
will do this in next pr